### PR TITLE
Test webui realm domains

### PR DESCRIPTION
**Pull request by celestian**
I prepare automation of some WebUI Realm Domains test.

This PR is continue of https://github.com/freeipa/freeipa/pull/1644

### CODE REVIEW

**Comment by celestian** [comment]
@felipevolpone , I would like to be sure that I will not damge PR CI again. Could you please add `re-run` label after you check that my configuration is right? Thank you.

**Comment by felipevolpone** [comment]
It is great now! Thank you! Just an extra info: if you want, you can change the line
```yml
test_suite: test_webui/
```
to
```yml
test_suite: test_webui/test_realmdomains.py
````

to only run the tests you're adding. This way the feedback will be faster.

**Comment by celestian** [comment]
@felipevolpone ,
I don't understand. PR CI removed `re-run` label, but how you can see, no results here. :-(

**Comment by felipevolpone** [comment]
Hi @celestian , you actually have a duplicated `ipaserver` section on it. Just remove it and it should be fine.

**Comment by celestian** [comment]
Fixed.

**Comment by pvoborni** [comment]
@celestian , based on two "failure" screenshots, I think that  2 or maybe all 3 failures are caused by a notification covering a space of element which needs some interaction. Basically same cause as in https://github.com/freeipa/freeipa/pull/1737 (there is more detailed description) solution might be the same.

**Comment by celestian** [comment]
Thanks, @pvoborni I fixed it with your proposal and force pushed it.

**Comment by celestian** [comment]
@Rezney It seems that it is GREEN now :-)
So, should I squash (is this the word?) my commits to.... 2, for example?

**Comment by celestian** [comment]
Last run (only on webUI realmd domains) passed. So I squashed it, I added support for all tests. And re-run it.

File: ipatests/test_webui/test_realmdomains.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
 # Authors:
 #   Petr Vobornik <pvoborni@redhat.com>
```
Comment (felipevolpone) [inline ipatests/test_webui/test_realmdomains.py]: Do we need to change this?

**Comment by felipevolpone** [comment]
The code and the test cases LGTM, but I'll leave the final ack to @pvoborni .

File: ipatests/test_webui/test_realmdomains.py (hunk 1 above)
Comment (celestian) [reply ipatests/test_webui/test_realmdomains.py]: We have https://github.com/freeipa/freeipa/blob/master/Contributors.txt
So... what about to delete it? @pvoborni , do you agree?

**Comment by pvoborni** [comment]
The way how the ticket is created mean that it will go only to master branch. Most of Web UI patches are being pushed also to ipa-4-6 branch to allow testing on system which won't have master (4.7). I guess that this PR should go to 4.6 as well. Right Petr? If so then, as all other patches which needs to be backported, it needs a ticket and commit messages must have a link to the ticket - this is useful for tracking which patch is where even if the backported patch is slightly modified. The future backport PR will also then depend on PR #1765

**Comment by celestian** [comment]
Rebased....

**Comment by celestian** [comment]
There some problems with space on disk. There should be fix on it, so I try rerun.

**Comment by celestian** [comment]
Timeout was to short.

**Comment by celestian** [comment]
@felipevolpone   I prolonged timeout to 10800, but it is not nice solution. How does the yaml key `test_suite` work? Is it possible to set something like:
```
test_suite: test_webui/test_dns, test_webui/test_group
```
If not, another solution is to group some of the test suite to the sub-folder and create more targets for PR CI.

**Comment by pvoborni** [comment]
`undo_multivalued` was already added to UI driver before this PR was completed so now pylint complains: 
```text
************* Module ipatests.test_webui.ui_driver
ipatests/test_webui/ui_driver.py:891: [E0102(function-redefined), UI_driver.undo_multivalued] method already defined line 841)
```

**Comment by felipevolpone** [comment]
@celestian yes, it's possible to have such thing. Another solution is to do what @Rezney did in his PR #1919 .

**Comment by celestian** [comment]
I removed duplicity in methods and I used .yaml configuration provided in #1919 .
Pushed again...

**Comment by pvoborni** [comment]
In #1919 the new test definition doesn't work yet, as can be observed by the failing tasks. Therefore, it won't add value here either till it is fixed.

**Comment by celestian** [comment]
@pvoborni. Thank you, I known that. Meanwhile we found another 3 tests which should be added. :-)

**Comment by celestian** [comment]
Nice, the way how to run webui test works!
Now, I need to add two missing tests.

**Comment by celestian** [comment]
Two missing tests added.
Only WebUI tests enabled.
Higher version of PR CI image.
Commit message needs update.

**Comment by celestian** [comment]
Please, what is wrong with travis here?

**Comment by celestian** [comment]
@felipevolpone  Could you please add ACK here?

**Comment by celestian** [comment]
Just note, new tests were green :-)

**Comment by Rezney** [comment]
master:

* 3c9810e9639095779e9b4f7c8bf37de3d4b8922c webui:tests: Add tests for realmd domains

**Pull request closed without merging**